Sprint 17

maksimbelov95 opened the pull request:
sprint_17

maksimbelov95 committed:
contextmenuadd

maksimbelov95 committed:
изменена локализация главното экрана

maksimbelov95 committed:
исправлена індекс метрика

maksimbelov95 committed:
добавление скриншот тесты

maksimbelov95 committed:
скриншот

maksimbelov95 committed:
в

maksimbelov95 committed:
добавлен функционал статистики

maksimbelov95 committed:
Добавлен градиент в проект

maksimbelov95 committed:
добавлен PlaceHolder в статистику

maksimbelov95 committed:
fix

maksimbelov95 committed:
fix edit

maksimbelov95 committed:
галочки

maksimbelov95 committed:
фикс

LizaIlinykh commented on Tracker/Statistic/Controller/StatisticViewController.swift:
```diff
@@ -3,58 +3,149 @@ import UIKit
+        return label
+    } ()
+    
+    override func viewWillAppear(_ animated: Bool) {
```
нужно вызвать метод родительского класса super.viewWillAppear(animated)

LizaIlinykh commented on Tracker/Statistic/Controller/StatisticViewController.swift:
```diff
@@ -3,58 +3,149 @@ import UIKit
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: "StatisticTableViewCell", for: indexPath) as! StatisticTableViewCell
```
рекомендую заменить принудительную операцию на optional binding

LizaIlinykh commented on Tracker/Statistic/Controller/StatisticViewController.swift:
```diff
@@ -3,58 +3,149 @@ import UIKit
+            UIColor.purple.cgColor
+        ]
+        gradientLayer.startPoint = CGPoint(x: 0.0, y: 0.5)
+        gradientLayer.endPoint = CGPoint(x: 1.0, y: 0.5)
```
лучше настраивать градиент в классе ячейки

LizaIlinykh commented on Tracker/TrackerCreate/Controllers/TrackerViewController.swift:
```diff
@@ -15,6 +18,8 @@ final class TrackerViewController: UIViewController {
     
     private var typeFilter: TypeFilter = .textAndDatePickerTrackers
     
+    private var savedFilter: String = "Все трекеры"
```
лучше создать перечисление , в котором кейсами будут типы фильтров

LizaIlinykh commented on Tracker/TrackerCreate/Controllers/TrackerViewController.swift:
```diff
@@ -115,6 +125,10 @@ final class TrackerViewController: UIViewController {
         trackersCollectionView.reloadData()
         categoriesStore.delegate = self
     }
```
между методами принято оставлять одну строку

maksimbelov95 committed:
1. На экране создания трекера клавиатура закрывается скроллом или кнопкой return, на экране создания категории клавиатура закрывается на тап по экрану
2. Добавлена заглушка на экран категории
3. Экран создания категорий переверстан
4. Исправлена копка готово
5. Тайте изменен
6. текст плейсхолдера в инпуте изменен
7. у цвета рамки у эмоджи добавлена прозрачность 0.3
8. у бордюра выбора цвета добавлена прозрачность 0.3
9. поставил таббар ypGray. хотя был вроде серый, специально созданный для таббара
10,13 Изменена логика работы фильтра
11. Изменена логика заглуши
12. Цвет шрифта лейбла с название тренера в темной теме теперь белый
